### DISCUSSION TITLE
How to set the "input_nodes " in HGTLoader? Two node types  Heterogeneous Graph？Only one node type can be set in "input_nodes"

### DISCUSSION THREAD

**Discussion by jasmineChouLujie** [pyg-team/pytorch_geometric#8869] [General]
![微信图片_20240206164043](https://github.com/pyg-team/pytorch_geometric/assets/44397416/ba543654-6e50-45ca-b7d4-23086e84b8df)
One node type is set as an example, how about two node types?

**Comment by rusty1s** [comment]
Yes, we currently only support a single input node type as "seed node" :( However, the the sampler will still sample neighbors of different node types.

**Comment by jasmineChouLujie** [comment]
Thank you. How can I handle two nodes type in HGT?

发自我的iPhone

**Reply by rusty1s** [reply]
`HGTConv` expects a dictionary of node types as features and a dictionary of edge types as graph structure. As such, it will perform heterogeneous message passing. If you are referring to `HGTLoader`, then the set of seed nodes is currently limited to a single node type.

**Comment by jasmineChouLujie** [comment]
Thank you for your help！

发自我的iPhone

**Comment by jykr** [comment]
Why do the heterogeneous data loaders start only from a single type? I just need a data loader for all nodes and it isn't seem to be supported.... Any suggestions when I'm interested in all node types?

**Reply by rusty1s** [reply]
It is quite complicated to support data loading from different node types, since a mini-batch would hold seed nodes from different types, which is challenging to process downstream. While I agree this is a nice feature to have, it needs a principled design to not be overly confusing to use.
